src/command_modules/azure-cli-sql/azure/cli/command_modules/sql/custom.py

```diff
@@ -910,3 +936,91 @@ def firewall_rule_update(
         resource_group_name=resource_group_name,
         start_ip_address=start_ip_address or instance.start_ip_address,
         end_ip_address=end_ip_address or instance.end_ip_address)
+
+#####
+#           sql server key
+#####
+
+def server_key_create(
+        client,
+        resource_group_name,
+        server_name,
+        key_id=None):
+    
+    key_name = get_server_key_name_from_uri(key_id)
+
+    return client.create_or_update(
+        resource_group_name=resource_group_name,
+        server_name=server_name,
+        key_name=key_name,
+        parameters=ServerKey(
+            server_key_type=ServerKeyType.azure_key_vault.value,
+            uri=key_id
+        )
+    )
+
+def server_key_get(
+        client,
+        resource_group_name,
+        server_name,
+        key_id):
+
+    key_name = get_server_key_name_from_uri(key_id)
+
+    return client.get(
+        resource_group_name=resource_group_name,
+        server_name=server_name,
+        key_name=key_name)
+
+def server_key_delete(
+        client,
+        resource_group_name,
+        server_name,
+        key_id):
+
+    key_name = get_server_key_name_from_uri(key_id)
+
+    return client.delete(
+        resource_group_name=resource_group_name,
+        server_name=server_name,
+        key_name=key_name)
+
+def get_server_key_name_from_uri(uri):
+
+    match = re.match('https(.)+\.vault(.)+\/keys\/[^\/]+\/[0-9a-zA-Z]+$', uri)
+
+    if match is None:
+        raise CLIError('The provided uri is invalid. Please provide a valid Azure Key Vault key id. For example: "https://YourVaultName.vault.azure.net/keys/YourKeyName/01234567890123456789012345678901"')
+
+    vault = uri.split('.')[0].split('/')[-1]
+    key = uri.split('/')[-2]
+    version = uri.split('/')[-1]
+    return '{}_{}_{}'.format(vault, key, version)
+
+#####
+#           sql server encryption-protector
+#####
+
+def encryption_protector_update(
+        client,
+        resource_group_name,
+        server_name,
+        server_key_type,
```

Comment:
So --server-key-type must be specified even if you only want to modify the key id, correct? This is not necessarily bad, just want to make sure this was a conscious choice.

Reply:
Correct. Even if the server is already using a Key Vault key as encryption protector, we still want them to provide server key type argument if they are simply changing to a different Key Vault key.